Assets/MRTK/Providers/LeapMotion/LeapMotionArticulatedHand.cs

```diff
@@ -0,0 +1,299 @@
+        /// <returns></returns>
+        private AttachmentPointFlags ConvertMRTKJointToLeapJoint(TrackedHandJoint joint)
+        {
+            switch (joint)
+            {
+                case TrackedHandJoint.Palm: return AttachmentPointFlags.Palm;
+                case TrackedHandJoint.Wrist: return AttachmentPointFlags.Wrist;
+
+                case TrackedHandJoint.ThumbProximalJoint: return AttachmentPointFlags.ThumbProximalJoint;
+                case TrackedHandJoint.ThumbDistalJoint: return AttachmentPointFlags.ThumbDistalJoint;
+                case TrackedHandJoint.ThumbTip: return AttachmentPointFlags.ThumbTip;
+
+                case TrackedHandJoint.IndexKnuckle: return AttachmentPointFlags.IndexKnuckle;
+                case TrackedHandJoint.IndexMiddleJoint: return AttachmentPointFlags.IndexMiddleJoint;
+                case TrackedHandJoint.IndexDistalJoint: return AttachmentPointFlags.IndexDistalJoint;
+                case TrackedHandJoint.IndexTip: return AttachmentPointFlags.IndexTip;
+
+                case TrackedHandJoint.MiddleKnuckle: return AttachmentPointFlags.MiddleKnuckle;
+                case TrackedHandJoint.MiddleMiddleJoint: return AttachmentPointFlags.MiddleMiddleJoint;
+                case TrackedHandJoint.MiddleDistalJoint: return AttachmentPointFlags.MiddleDistalJoint;
+                case TrackedHandJoint.MiddleTip: return AttachmentPointFlags.MiddleTip;
+
+                case TrackedHandJoint.RingKnuckle: return AttachmentPointFlags.RingKnuckle;
+                case TrackedHandJoint.RingMiddleJoint: return AttachmentPointFlags.RingMiddleJoint;
+                case TrackedHandJoint.RingDistalJoint: return AttachmentPointFlags.RingDistalJoint;
+                case TrackedHandJoint.RingTip: return AttachmentPointFlags.RingTip;
+
+                case TrackedHandJoint.PinkyKnuckle: return AttachmentPointFlags.PinkyKnuckle;
+                case TrackedHandJoint.PinkyMiddleJoint: return AttachmentPointFlags.PinkyMiddleJoint;
+                case TrackedHandJoint.PinkyDistalJoint: return AttachmentPointFlags.PinkyDistalJoint;
+                case TrackedHandJoint.PinkyTip: return AttachmentPointFlags.PinkyTip;
+
+                // Metacarpals are not included in AttachmentPointFlags
+                default: return AttachmentPointFlags.Wrist;
+            }
+        }
+
+        public void UpdateState()
+        {
+            // Populate the jointPoses 
+            for (int i = 0; i < jointCount; i++)
+            {
+                TrackedHandJoint handJoint = (TrackedHandJoint)i;
+                TryGetJoint(handJoint, out MixedRealityPose pose);
+
+                if (!jointPoses.ContainsKey(handJoint))
+                {
+                    jointPoses.Add(handJoint, pose);
+                }
+                else
+                {
+                    jointPoses[handJoint] = pose;
+                }
```

Comment:
nit: I'm pretty sure dictionaries are fine with this syntax for both cases, so we don't have to check that it contains the key and explicitly call `Add`. Also might be good to check that `TryGetJoint` succeeded

```suggestion
                if (TryGetJoint(handJoint, out MixedRealityPose pose))
                {
                    jointPoses[handJoint] = pose;
                }
```

Reply:
Added